### ISSUE TITLE
Skirmish mech bay shows empty loadout for mech

### ISSUE THREAD

**Issue by Sheep-y** [Sheep-y/BattleTech_Turbine#1]
Happens until you actually edit a mech. Then everything becomes normal. Very interesting.

```
NullReferenceException: Object reference not set to an instance of an object
  at BattleTech.MechStatisticsRules.CalculateHeatEfficiencyStat (BattleTech.MechDef mechDef, System.Single& currentValue, System.Single& maxValue) [0x00000] in <filename unknown>:0 
  at BattleTech.UI.MechBayMechInfoWidget.SetStat (BattleTech.UI.LanceStat stat, BattleTech.UI.StatisticValidationFunction calcStat) [0x00000] in <filename unknown>:0 
  at BattleTech.UI.MechBayMechInfoWidget.SetStats () [0x00000] in <filename unknown>:0 
  at BattleTech.UI.MechBayMechInfoWidget.SetData (BattleTech.SimGameState sim, BattleTech.UI.MechBayPanel mechBay, BattleTech.Data.DataManager dataManager, BattleTech.UI.MechBayMechUnitElement mechElement, Boolean useNoMechOverlay, Boolean useRepairButton) [0x00000] in <filename unknown>:0 
  at BattleTech.UI.SkirmishMechBayPanel.SelectMech (BattleTech.MechDef mech) [0x00000] in <filename unknown>:0 
  at BattleTech.UI.SkirmishMechBayPanel.OnButtonClicked (IMechLabDraggableItem item) [0x00000] in <filename unknown>:0 
  at BattleTech.UI.MechBayMechStorageWidget.OnButtonClicked (IMechLabDraggableItem item) [0x00000] in <filename unknown>:0 
  at BattleTech.UI.MechBayMechUnitElement.OnButtonClicked () [0x00000] in <filename unknown>:0 
  at UnityEngine.Events.InvokableCall.Invoke (System.Object[] args) [0x00000] in <filename unknown>:0 
  at UnityEngine.Events.InvokableCallList.Invoke (System.Object[] parameters) [0x00000] in <filename unknown>:0 
  at UnityEngine.Events.UnityEventBase.Invoke (System.Object[] parameters) [0x00000] in <filename unknown>:0 
  at UnityEngine.Events.UnityEvent.Invoke () [0x00000] in <filename unknown>:0 
  at BattleTech.UI.HBSToggle.OnPointerUp (UnityEngine.EventSystems.PointerEventData eventData) [0x00000] in <filename unknown>:0 
```

**Comment by Sheep-y** [comment]
How it happens:

1. `MechStatisticsRules.Combat` is null.
2. `CalculateHeatEfficiencyStat` calls  `CombatGameConstants.CreateFromSaved(UnityGameInstance.BattleTechGame)`
3. `CreateFromSaved(game)` calls `new CombatGameConstants(game)`
4. The `CombatGameConstants` is initialised with a default `MoveConstants` which has null `MoveTable`.
5. The `LoadFromManifest` somehow failed to fill it in.

Next step: find how is CombatGameConstants loaded and why is it not loaded.

**Comment by Sheep-y** [comment]
Stacktrace of bugged and non-bugged load. Vanilla game loads the constant three times but Turbine do it only once.

[bugged.txt](https://github.com/Sheep-y/BattleTech_Turbine/files/2249752/bugged.txt)
[vanilla.txt](https://github.com/Sheep-y/BattleTech_Turbine/files/2249753/vanilla.txt)

**Comment by Sheep-y** [comment]
Don't have time to got to the bottom. Instead it is fixed in 1.0.3 by caching the json text and in 1.0.4 by caching the constant object. So far havn't got other reports.
